python/ql/src/Security/CWE-327/Ssl.qll

```diff
@@ -0,0 +1,186 @@
+import python
+import semmle.python.ApiGraphs
+import semmle.python.dataflow.new.internal.Attributes as Attributes
+import TlsLibraryModel
+
+class SSLContextCreation extends ContextCreation {
+  override CallNode node;
+
+  SSLContextCreation() { this = API::moduleImport("ssl").getMember("SSLContext").getACall() }
+
+  override DataFlow::CfgNode getProtocol() {
+    result.getNode() in [node.getArg(0), node.getArgByName("protocol")]
+  }
```

Comment:
How do you handle modeling the default protocol? I would have expected a `not exists(protocolArg()) and result in [...]`

Reply:
It is handled in `UnspecificSSLContextCreation` where we reinterpret the `SSLContextCreation` as a `ProtocolUnrestriction`, except I had forgotten to do that.

The route you suggest might be simpler, but then we need to switch the return type to string, currently there is no data-flow node to report if there is no argument given. What I did was a smaller diff.